credoai/modules/model_assessments/nlp_generator.py

```diff
@@ -0,0 +1,496 @@
+                temp["value"] = temp["response"].apply(
+                    lambda x: self.assess_with_perspective(
+                        x, assessment_attribute, rpm_limit
+                    )
+                )
+                temp["assessment_attribute"] = assessment_attribute
+                dfrunst_assess = dfrunst_assess.append(temp)
+        else:
+            raise Exception(
+                "Assessment cannot be completed due to how assessment is configured in the provided assessment_config."
+            )
+
+        self.raw_results = dfrunst_assess
+
+        return self
+
+    def _get_prompts(self, prompt_dataset):
+        """Load the prompts dataset from a csv file as a dataframe
+
+        Parameters
+        ----------
+        prompt_dataset : str
+            One of the following:
+                Name of a builtin prompt dataset. Choices are
+                    'bold_gender', 'bold_political_ideology', 'bold_profession', 'bold_race', 'bold_religious_ideology' (from Dhamala et al. 2021)
+                    'realtoxicityprompts_1000', 'realtoxicityprompts_challenging_20', 'realtoxicityprompts_challenging_100', 'realtoxicityprompts_challenging' (from Gehman et al. 2020)
+                Path of your own prompts csv file with columns 'group', 'subgroup', 'prompt'
+
+        Returns
+        -------
+        pandas.dataframe
+            prompts dataframe with columns
+            Schema: ['group', 'subgroup', 'prompt']
+
+        Raises
+        ------
+        Exception
+            Occurs if the provided prompts dataset csv file is not a valid file
+        Exception
+            Occurs if the prompts dataset cannot be loaded
+        """
+        if prompt_dataset.split(".")[-1] == "csv":
+            df = pd.read_csv(prompt_dataset)
+            cols_required = ["group", "subgroup", "prompt"]
+            cols_given = list(df.columns)
+            if set(cols_given) != set(cols_required):
+                cols_required_str = ", ".join(cols_required)
+                raise Exception(
+                    "The provided prompts dataset csv file is not a valid file. Ensure it has all and only the following columns: "
+                    + cols_required_str
+                )
+
+        elif prompt_dataset in PROMPTS_PATHS:
+            prompts_path = get_data_path(PROMPTS_PATHS[prompt_dataset])
+            df = pd.read_csv(prompts_path)
+
+        else:
+            builtin_prompts_names = list(PROMPTS_PATHS.keys())
+            builtin_prompts_names = ", ".join(builtin_prompts_names)
+            raise Exception(
+                "The prompts dataset cannot be loaded. Ensure the provided prompt_dataset value is either a path to a valid csv file"
+                + " or name of one of the builtin datasets (i.e."
+                + builtin_prompts_names
+                + "). You provided "
+                + prompt_dataset
+            )
+
+        return df
+
+    def build_perspective_client(self, api_key):
```

Comment:
This (and other perspective functions) look like they should start with "_"

Reply:
private function now are all updated to reflect this